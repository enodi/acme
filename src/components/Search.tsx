import React from "react";

const Search: React.FC<{ filterByFullname: any }> = ({ filterByFullname }) => (
  <input
    placeholder="Search for a new chat..."
    className="search"
    onChange={filterByFullname}
  />
);

export default Search;
