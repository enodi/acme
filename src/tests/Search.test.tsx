import React from "react";
import { shallow } from "enzyme";

import Search from "../components/Search";
import { selectedItem } from "./fixtures";

let filterByFullname: (event: React.ChangeEvent<HTMLInputElement>) => void,
  wrapper: any;
beforeEach(() => {
  filterByFullname = jest.fn();
  wrapper = shallow(<Search filterByFullname={filterByFullname} />);
});

test("should render Search component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should handle search change", () => {
  const value = selectedItem.firstName;
  wrapper.find("input").simulate("change", value);
  expect(filterByFullname).toHaveBeenLastCalledWith(value);
});
