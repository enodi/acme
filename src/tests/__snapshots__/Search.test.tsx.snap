// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search component correctly 1`] = `
<input
  className="search"
  onChange={[MockFunction]}
  placeholder="Search for a new chat..."
/>
`;
